apps/game_backend/lib/game_backend/rewards.ex

```diff
@@ -43,4 +44,73 @@ defmodule GameBackend.Rewards do
         insert_afk_reward_rate(%{user_id: user_id, currency_id: currency_id, rate: rate_increment})
     end
   end
+
+  @doc """
+  Receives a user.
+  Returns the next daily reward for the given user.
+  Or the first claim if never claimed at all.
+  """
+  def claim_daily_reward(%GameBackend.Users.User{last_daily_reward_claim_at: nil}),
+    do: claim_first_reward(Utils.get_daily_rewards_config())
+
+  def claim_daily_reward(user) do
+    daily_reward_config = Utils.get_daily_rewards_config()
+    yesterday = DateTime.utc_now() |> Date.add(-1)
+
+    case Date.compare(user.last_daily_reward_claim_at, yesterday) do
+      :eq -> claim_next_reward(daily_reward_config, user.last_daily_reward_claim)
+      _ -> claim_first_reward(daily_reward_config)
+    end
+  end
+
+  defp claim_first_reward(daily_reward_config) do
+    case Map.get(daily_reward_config, "1") do
+      nil -> {:error, :invalid_reward}
+      first_daily_reward -> {:ok, first_daily_reward}
+    end
+  end
+
+  defp claim_next_reward(daily_reward_config, current_reward) do
+    case Map.get(daily_reward_config, current_reward) do
+      nil -> {:error, :invalid_reward}
+      next_daily_reward -> {:ok, next_daily_reward}
+    end
+  end
+
+  @doc """
+  Receives a user.
+  Returns {:ok, can_claim} if the user claimed today already or never claimed at all.
+  """
+  def user_claimed_today(%GameBackend.Users.User{last_daily_reward_claim_at: nil}), do: {:ok, :can_claim}
+
+  def user_claimed_today(user) do
+    now = DateTime.utc_now()
+
+    case Date.compare(user.last_daily_reward_claim_at, now) do
+      :eq -> {:ok, :can_claim}
+      _ -> {:error, :already_claimed}
+    end
+  end
+
+  defp user_in_daily_rewards_streak?(%GameBackend.Users.User{last_daily_reward_claim_at: nil}), do: false
+
+  defp user_in_daily_rewards_streak?(user) do
+    today = DateTime.utc_now()
+    yesterday = today |> Date.add(-1)
+
+    Date.compare(user.last_daily_reward_claim_at, today) == :eq or
+      Date.compare(user.last_daily_reward_claim_at, yesterday) == :eq
+  end
+
+  def mark_user_daily_rewards_as_completed(daily_rewards_config, user) do
```

Comment:
Not sure what is wrong, but something is wrong here cause I get the following (or I missunderstood something)

```
>curl http://localhost:4001/users/bd984c53-ca09-418e-aeb7-9ba6cbf58f54/get_daily_reward_status | jq
{
  "1": "not claimed",
  "2": "not claimed",
  "3": "not claimed",
  "4": "not claimed",
  "5": "not claimed",
  "6": "not claimed",
  "7": "not claimed"
}

> curl http://localhost:4001/users/bd984c53-ca09-418e-aeb7-9ba6cbf58f54/claim_daily_reward

> curl http://localhost:4001/users/bd984c53-ca09-418e-aeb7-9ba6cbf58f54/get_daily_reward_status | jq
{
  "1": "not claimed", ## <- this should be claimed right?
  "2": "claimed",       ## <- This and all other should be not claimed right?
  "3": "claimed",
  "4": "claimed",
  "5": "claimed",
  "6": "claimed",
  "7": "claimed"
}
```

Reply:
This should be fixed now!